src/api/v1/stripe_donate.go

```diff
@@ -120,33 +101,133 @@ func redeemStripePayment(c echo.Context) error {
 		return echo.NewHTTPError(http.StatusBadRequest, "Bad Payment Amount: Payment "+body.ID+" totals "+strconv.FormatInt(payment.PaymentIntent.Amount, 10)+", expected 500 or more")
 	}
 
+	// Now that we are interacting with the DB we should lock
+	donationLock.Lock()
```

Comment:
Needs testing - is it ok to lock this in one place and then unlock it in a child goroutine? e.g. like i currently do in `redeemStripePayment`?